spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/rdd/reader/PrefetchingResultSetIterator.scala

```diff
@@ -8,16 +12,27 @@ import com.datastax.driver.core.{Row, ResultSet}
   * @param resultSet result set obtained from the Java driver
   * @param prefetchWindowSize if there are less than this rows available without blocking,
   *                           initiates fetching the next page
+  * @param timer a Codahale timer to optionally gather the metrics of fetching time
   */
-class PrefetchingResultSetIterator(resultSet: ResultSet, prefetchWindowSize: Int) extends Iterator[Row] {
+class PrefetchingResultSetIterator(resultSet: ResultSet, prefetchWindowSize: Int, timer: Option[Timer] = None) extends Iterator[Row] {
 
   private[this] val iterator = resultSet.iterator()
 
   override def hasNext = iterator.hasNext
 
   private[this] def maybePrefetch(): Unit = {
-    if (!resultSet.isFullyFetched && resultSet.getAvailableWithoutFetching < prefetchWindowSize)
-      resultSet.fetchMoreResults()
+    if (!resultSet.isFullyFetched && resultSet.getAvailableWithoutFetching < prefetchWindowSize) {
+      val t0 = System.nanoTime()
+      val future: ListenableFuture[Void] = resultSet.fetchMoreResults()
+      if (timer.isDefined)
+        Futures.addCallback(future, new FutureCallback[Void] {
```

Comment:
I feel a little strange about us recording request latencies and the driver also doing it at the same time (http://www.datastax.com/drivers/java/2.1/com/datastax/driver/core/Metrics.htm). I understand this lets us separate out read's from writes but maybe it would be beneficial if we try to get the separation in upstream in the Java Driver rather than redoing the work in the connector?